### DISCUSSION TITLE
Improvements can be made on the internal webview feature

### DISCUSSION THREAD

**Discussion by FerasK0** [robertodoering/harpy#771] [Ideas]
Hi Robert, long time no see :)

Anyway I liked opening Twitter links with harpy and the link preview feature, but like always there's room of improvement.

-If the internal webview is customizable then take these into consideration-

First: an exit button, it'd really be a good idea to implement it because if your went deep into a couple of pages you have to swipe or press back a couple times to get back to harpy.

Second: a share button, so let's say i.e I read an article about something and I found it interesting and I want to share it, I can't unless there's a share button somewhere in that page, or I have to quit the page and press and hold the link to copy.

Third: in an outside note, the link preview hold interaction should match/behave like the link interaction if you get what I mean.

Thanks for the amazing efforts again, keep it up! <3

**Comment by dnestico** [comment]
I agree it should be a proper web view like google app or Twitter app, it also refuses to exit sometimes she have to restart app

**Comment by dnestico** [comment]
Link preview doesn't seem to work with "Twitter for advertisers" tweets ![Screenshot_20220929-092533.png](https://user-images.githubusercontent.com/3034555/193044185-14c74340-12e1-41c3-a903-240e6d09c25a.png)

**Comment by robertodoering** [comment]
Thanks! I've added some issues with your feedback: #780 #781
